tests/testthat/test-oauth.R

```diff
@@ -53,6 +53,75 @@ test_that("oauth_encode1 works", {
   expect_equal(orig_string, restored_string)
 })
 
+test_that("oauth2.0 authorize url appends query params", {
+  app <- oauth_app("x", "y", "z")
+  scope <- NULL
+  query_extra <- list(
+    foo = "bar"
+  )
+  authURL <- oauth2.0_authorize_url(
+    endpoint = oauth_endpoints("google"),
+    app = app,
+    scope = scope,
+    query_extra = query_extra
+  )
+  
+  url <- parse_url(authURL)
+  expect_equal(url$query$foo, "bar")
+})
+
+test_that("oauth2.0 authorize url with empty input does not append empty query params", {
```

Comment:
I still don't understand what you're trying to test here.

Reply:
basically that providing empty input to `query_extra` doesn't break things.
I've replaced the test with something that's hopefully clearer.